lms/djangoapps/teams/serializers.py

```diff
@@ -0,0 +1,140 @@
+"""
+Defines serializers used by the Teams API.
+"""
+
+from django.contrib.auth.models import User
+from rest_framework import serializers
+from openedx.core.lib.api.fields import ExpandableField
+from .models import CourseTeam, CourseTeamMembership
+from openedx.core.djangoapps.user_api.serializers import UserSerializer
+
+
+class CollapsedUserSerializer(serializers.HyperlinkedModelSerializer):
+    """
+    Serializes users in a collapsed format, with just a username and url.
+    """
+    url = serializers.HyperlinkedIdentityField(view_name='accounts_api', lookup_field='username')
+
+    class Meta(object):
+        """
+        Defines meta information for the ModelSerializer.
+        """
+        model = User
+        fields = ("username", "url")
+        read_only_fields = ("username",)
+
+
+class CollapsedCourseTeamSerializer(serializers.HyperlinkedModelSerializer):
+    """
+    Serializes CourseTeams in a collapsed format, with just an id and url.
+    """
+    id = serializers.CharField(source='team_id', read_only=True)  # pylint: disable=invalid-name
+    url = serializers.HyperlinkedIdentityField(view_name='teams_detail', lookup_field='team_id')
+
+    class Meta(object):
+        """
+        Defines meta information for the ModelSerializer.
+        """
+        model = CourseTeam
+        fields = ("id", "url")
+
+
+class UserMembershipSerializer(serializers.ModelSerializer):
+    """
+    Serializes CourseTeamMemberships with only information about the user and date_joined, for listing team members.
+    """
+    user = ExpandableField(
+        collapsed_serializer=CollapsedUserSerializer(),
+        expanded_serializer=UserSerializer(),
+    )
+
+    class Meta(object):
+        """
+        Defines meta information for the ModelSerializer.
+        """
+        model = CourseTeamMembership
+        fields = ("user", "date_joined")
+        read_only_fields = ("date_joined",)
```

Comment:
It effectively is. I wrote the ExpandableField class, and it only implements serialization, so it would be impossible to write to the user field.